terraform/state.go

```diff
@@ -644,6 +653,14 @@ type ResourceState struct {
 	// However, in pathological cases, it is possible for the number
 	// of instances to accumulate.
 	Tainted []*InstanceState `json:"tainted,omitempty"`
+
+	// Deposed is used in the mechanics of CreateBeforeDestroy: the existing
+	// Primary is Deposed to get it out of the way for the replacement Primary to
+	// be created by Apply. If the replacement Primary creates successfully, the
+	// Deposed instance is cleaned up. If there were problems creating the
+	// replacement, we mark the replacement as Tainted and Undepose the former
+	// Primary.
```

Comment:
I would just add here for future docs sake that for all intents and purposes, a deposed instance can be treated like a tainted instance in that it must be destroyed.